src/NuGet.Core/NuGet.Build.Tasks/GetRestorePrunedPackageReferencesTask.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace NuGet.Build.Tasks
+{
+    public class GetRestorePrunePackageReferencesTask : Task
+    {
+        /// <summary>
+        /// Full path to the msbuild project.
+        /// </summary>
+        [Required]
+        public string ProjectUniqueName { get; set; }
+
+        [Required]
+        public ITaskItem[] PrunePackageReferences { get; set; }
+
+        /// <summary>
+        /// Target frameworks to apply this for. If empty this applies to all.
+        /// </summary>
+        public string TargetFrameworks { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] RestoreGraphItems { get; set; }
+
+        public override bool Execute()
+        {
+            if (PrunePackageReferences.Length == 0) return true;
+
+            var entries = new List<ITaskItem>(PrunePackageReferences.Length);
+            var seenIds = new HashSet<string>(PrunePackageReferences.Length, StringComparer.Ordinal);
```

Comment:
Does this need to use a case insensitive comparer?

Reply:
Technically yes, but fortunately there's another check that "hid" this bug.

I had some test in the "full implementation" that validated that, which worked because `MSBuildRestoreUtility` itself handles the correct casing. 

I'll make a change and push some tests.

Reply:
I think you should add a comment explaining, just in case someone runs a code analysis some day and tries to be too smart.